src/plugins/query_enhancements/public/datasets/s3_type.test.ts

```diff
@@ -142,36 +142,40 @@ describe('s3TypeConfig', () => {
     it('should fetch data sources for unknown type', async () => {
       mockSavedObjectsClient.find = jest.fn().mockResolvedValue({
         savedObjects: [
-          { id: 'ds1', attributes: { title: 'DataSource 1', dataSourceEngineType: 'OpenSearch' } },
+          { id: 'ds1', attributes: { title: 'DataSource 1', dataSourceVersion: '3.0' } },
         ],
       });
 
       const result = await s3TypeConfig.fetch(mockServices as IDataPluginServices, [
         { id: 'unknown', title: 'Unknown', type: 'UNKNOWN' },
       ]);
 
-      expect(result.children).toHaveLength(1); // Including DEFAULT_DATA.STRUCTURES.LOCAL_DATASOURCE
+      expect(result.children).toHaveLength(1);
       expect(result.children?.[0].title).toBe('DataSource 1');
       expect(result.hasNext).toBe(true);
     });
 
-    it('should filter out OpenSearch Serverless data sources', async () => {
+    it('should filter out data sources with versions lower than 1.0.0', async () => {
       mockSavedObjectsClient.find = jest.fn().mockResolvedValue({
         savedObjects: [
-          { id: 'ds1', attributes: { title: 'DataSource 1', dataSourceEngineType: 'OpenSearch' } },
+          { id: 'ds1', attributes: { title: 'DataSource 1', dataSourceVersion: '1.0' } },
           {
             id: 'ds2',
-            attributes: { title: 'DataSource 2', dataSourceEngineType: 'OpenSearch Serverless' },
+            attributes: { title: 'DataSource 2', dataSourceVersion: '' },
+          },
+          { id: 'ds3', attributes: { title: 'DataSource 3', dataSourceVersion: '2.17.0' } },
+          {
+            id: 'ds4',
+            attributes: { title: 'DataSource 4', dataSourceVersion: '.0' },
           },
```

Comment:
can we add some test to check invalid semver version and semver version with snapshot/alpha etc?

Reply:
What happens on an invalid version? I assume coerce is just not touching it and it fails satisfies? Data source 2 is an example of an invalid version, but the truthy check on coercedVersion handles that case (I assume).

Docs looks like coerce should handle second case so don't think that's a big issue.

Reply:
Confirmed an invalid version returns null locally, so lgtm